### PR TITLE
Plugin clean up and DSO plugin loading

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
@tribal-tec @bilgili @jafyvilla Please review

File: doc/RelNotes.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ and documentation changes over Lunchbox 1.9:
 * lunchbox::URI parsing support according to RFC3986
 * lunchbox::PersistentMap interface for persistent key-value stores
 * lunchbox::getHostname and lunchbox::getExecutablePath functions
-* lunchbox::IOPluginFactory, a mechanism to register input/ouput plugins has
+* lunchbox::PluginFactory, a mechanism to register input/ouput plugins has
```
Comment (tribal-tec) [inline doc/RelNotes.md]: remove input/output then?

File: lunchbox/plugin.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+ * @code
+ * DerivedPluginClass( const InitDataT& initData );
+ * @endcode
+ *
+ * PluginT must also implement the following method to be registered:
+ * @code
+ * static bool handles( const InitDataT& initData );
+ * @endcode
+ *
+ * @version 1.10.0
```
Comment (tribal-tec) [inline lunchbox/plugin.h]: version bump?

**Comment by tribal-tec** [comment]
+1

File: doc/RelNotes.md (hunk 1 above)
Comment (eile) [inline doc/RelNotes.md]: done

File: lunchbox/plugin.h (hunk 2 above)
Comment (eile) [inline lunchbox/plugin.h]: done

**Pull request merged**